lib/middleware/init.js

```diff
@@ -31,9 +35,14 @@ exports.init = function(app){
     req.res = res;
     res.req = req;
     req.next = next;
+    if (isHttp2Supported && req instanceof http2Request) {
```

Comment:
This should be checked by looking at `req. httpVersionMajor ` instead.

Reply:
Not sure about this: 
What if https://github.com/spdy-http2/node-spdy/issues/316 is indeed fixed yet node-spdy still uses the http1 IncomingMessage?

Reply:
This actually seems fine now that I have fresh eyes. I don't know why I thought it was an issue anyway, I guess just stared at the code too long.